framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -103,64 +114,51 @@ InterfaceMapConstPtr ServiceReferenceBasePrivate::GetPrototypeService(const Bund
 
 std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bundle)
 {
-  auto s = ExtractInterface(GetServiceInterfaceMap(bundle), interfaceId);
-  if (!s)
-  {
-    registration->Lock(), --registration->dependents[bundle];
-  }
-  return s;
+  return ExtractInterface(GetServiceInterfaceMap(bundle), interfaceId);
```

Comment:
The impl of `ExtractInterface` should still be improved. When it gets a null shared_ptr, it will dereference it. I think this is were the code originally crashed due to the race in `GetServiceInterfaceMap`.